core/src/main/scala/zio/logging/Logger.scala

```diff
@@ -43,15 +44,16 @@ trait Logger[-A] { self =>
   /**
    * Logs the specified element at the error level with cause.
    */
-  def error(line: => A, cause: Cause[Any]): UIO[Unit] =
-    self.locally(LogAnnotation.Cause(Some(cause))) {
+  def error[E: CauseToThrowable](line: => A, cause: Cause[E]): UIO[Unit] =
```

Comment:
My intention was to be able to write well typed `E => Throwable` transformations because `E` gets erased when it's put into the annotation. So this is the last place where it can be captured.

Reply:
we can change annotation or create new one for this purpose.

Reply:
If this is required, and I am not sure that it is (see question on the PR), then the idiomatic solution is to require `Cause[Throwable]` here, rather than to accept `Cause[E]` and a function `E => Throwable`.

If the user can go from `E` to `Throwable`, then they can just `map` over the `Cause[E]` to turn it to `Cause[Throwable]` before passing it into this function.